src/libs/actions/OnyxDerived/types.ts

```diff
@@ -0,0 +1,22 @@
+import type {OnyxValue} from 'react-native-onyx';
+import type {NonEmptyTuple, ValueOf} from 'type-fest';
+import type {OnyxKey} from '@src/ONYXKEYS';
+import type ONYXKEYS from '@src/ONYXKEYS';
+
+/**
+ * A derived value configuration describes:
+ *  - a tuple of Onyx keys to subscribe to (dependencies),
+ *  - a compute function that derives a value from the dependent Onyx values.
+ *    The compute function receives a single argument that's a tuple of the onyx values for the declared dependencies.
+ *    For example, if your dependencies are `['report_', 'account'], then compute will receive a [OnyxCollection<Report>, OnyxEntry<Account>]
+ */
+type OnyxDerivedValueConfig<Key extends ValueOf<typeof ONYXKEYS.DERIVED>, Deps extends NonEmptyTuple<Exclude<OnyxKey, Key>>> = {
+    key: Key;
+    dependencies: Deps;
+    compute: (args: {
+        -readonly [Index in keyof Deps]: OnyxValue<Deps[Index]>;
+    }) => OnyxValue<Key>;
```

Comment:
random thought while testing: we could consider adding `currentValue` as second param. This might be helpful for returning a value instantly if we are sure it doesn't make sense to recompute it. Already found two places might benefit from this:
- `conciergeChatReportID`
- `findSelfDMReportID` function in `ReportUtils`

Reply:
updated!